src/encoded/submission_status.py

```diff
@@ -71,7 +71,13 @@ def get_submission_status(context, request):
     }
 
 
-def add_submission_status_search_filters(search_params, filter):
+def add_submission_status_search_filters(search_params, filter, fileSetSearchId):
+    # Direct search by submitted_id takes precendence
+    if fileSetSearchId:
+        targeted_prop = "accession" if is_accession(fileSetSearchId) else "submitted_id"
+        search_params[targeted_prop] = fileSetSearchId
+        return
```

Comment:
This function makes big assumptions about what is passed to it, I would certainly mention this in docstring